### PR TITLE
Dump battle scripts and several rfu3 strings

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: asm/macros/battle_script.inc [hunk 1]
```diff
@@ -1022,207 +1042,337 @@
...
 	.byte 0xf6
 	.endm
 
-	.macro activesidesomething
+	.macro finishturn
 	.byte 0xf7
 	.endm
 
-	.macro atkf8 bank
+	.macro trainerslideout param0:req
```
Comment (jiangzhengwenjz) [inline asm/macros/battle_script.inc]: firered doesn't have atkf8

File: data/battle_message.s [hunk 2]
```diff
@@ -2173,48 +2174,270 @@ gUnknown_83FDF3C:: @ 83FDF3C
...
-	.2byte 0x017e, 0x017e, 0x0180, 0x0181
+	.2byte STRINGID_ATTACKMISSED
+	.2byte STRINGID_PKMNPROTECTEDITSELF
+	.2byte STRINGID_PKMNAVOIDEDATTACK
+	.2byte STRINGID_AVOIDEDDAMAGE
+	.2byte STRINGID_PKMNMAKESGROUNDMISS
+
+gNoEscapeStringIds::
+	.2byte STRINGID_CANTESCAPE
+	.2byte STRINGID_DONTLEAVEBIRCH
```
Comment (jiangzhengwenjz) [inline data/battle_message.s]: Lol. R/S leftover?

File: data/battle_scripts_2.s [hunk 3]
```diff
@@ -1,28 +1,238 @@
...
 
-gBattlescriptsForUsingItem:: @ 81D99E4
-	.incbin "baserom.gba", 0x1D99E4, 0x18
+gBattlescriptsForUsingItem::
+	.4byte BattleScript_PlayerUseItem
+	.4byte BattleScript_AIuseFullRestoreOrHpHeal
+	.4byte BattleScript_AIuseFullRestoreOrHpHeal
+	.4byte BattleScript_AIuseStatRestore
+	.4byte BattleScript_AIuseXstat
+	.4byte BattleScript_AIuseGuardSpec
```
Comment (jiangzhengwenjz) [inline data/battle_scripts_2.s]: use -> Use

File: data/battle_scripts_2.s [hunk 4]
```diff
@@ -1,28 +1,238 @@
...
 
-gUnknown_81D9A88:: @ 81D9A88
-	.incbin "baserom.gba", 0x1D9A88, 0xB
+BattleScript_OldManThrowBall::
+	printstring STRINGID_OLDMANUSEDITEM
+	handleballthrow
 
-BattleScript_ShakeBallThrow:: @ 81D9A93
-	.incbin "baserom.gba", 0x1D9A93, 0x2E
+BattleScript_PokeDudeThrowBall::
```
Comment (jiangzhengwenjz) [inline data/battle_scripts_2.s]: inconsistent with line 99. I use Pokedude instead of PokeDude on my branch

File: data/link_rfu_3.s [hunk 5]
```diff
@@ -208,148 +208,511 @@ gUnknown_8457C20:: @ 8457C20
 	.incbin "baserom.gba", 0x457C20, 0x28
 
 gUnknown_8457C48:: @ 8457C48
-	.incbin "baserom.gba", 0x457C48, 0x5C
+	.string "The other TRAINER showed\nyou their TRAINER CARD.\pWould you like to show your\nTRAINER CARD?$"
```
Comment (jiangzhengwenjz) [inline data/link_rfu_3.s]: Maybe you can split it to multi-line?

File: data/battle_scripts_2.s (hunk 3 above)
Comment (PikalaxALT) [reply data/battle_scripts_2.s]: AIUse or AiUse?

File: data/battle_message.s (hunk 2 above)
Comment (PikalaxALT) [reply data/battle_message.s]: Yep

File: include/battle_scripts.h [hunk 6]
```diff
@@ -226,9 +226,9 @@ extern const u8 BattleScript_MoveEffectRecoil[];
 extern const u8 BattleScript_FlushMessageBox[];
 extern const u8 BattleScript_SpikesOnFaintedBattler[];
 extern const u8 BattleScript_GhostBallDodge[];
-extern const u8 gUnknown_81D9A88[];
-extern const u8 gUnknown_81D91A1[]; // knizz: silph scope unveil
-extern const u8 gUnknown_81D88D7[];
+extern const u8 BattleScript_OldMan_Pokedude_CaughtMessage[];
+extern const u8 BattleScript_SilphScopeUnveiled[]; // knizz: silph scope unveil
```
Comment (jiangzhengwenjz) [inline include/battle_scripts.h]: The comment should be removed

**Pull request merged**